v2/swift/main.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  main.swift
+//  YelpAPISample
+//
+
+import Foundation
+
+// Fill in your API keys here from
+// https://www.yelp.com/developers/manage_api_keys
+let client = YLPClient(consumerKey: "",
+                       consumerSecret: "",
+                       token: "",
+                       tokenSecret: "")
+
+// Create a dispatch group to wait for search results
+let group = dispatch_group_create()
+dispatch_group_enter(group)
```

Comment:
Great question @gsheld! I've never actually used `dispatch_semaphore`, always just `dispatch_group`. From discussion in IRC, these were the proposed guidelines: 
> dispatch_groups are best for when you have multiple tasks you need to ensure all finish before continuing execution
> dispatch_semaphores are best for when you have a shared resource you want to ensure is accessed safely

If we're going from that definition, although there's only one task executing here, it seems like `dispatch_group` might semantically be the better fit. Let me know what you think.